### ISSUE TITLE
Unable to install using Pyspark / ivy

### ISSUE THREAD

**Issue by julwin** [exasol/spark-connector#128]
Pyspark version: 3.3.1

The error occurs when trying to add Exasol connector as dependencye.g.:
` com.exasol:spark-connector_2.13:1.2.3-spark-3.3.1`

Error: 
```
Exception in thread "main" java.lang.RuntimeException: [unresolved dependency: com.exasol#spark-connector_2.13;1.2.3-spark-3.3.1: not found
```

io problem while parsing ivy file: https://repo1.maven.org/maven2/com/exasol/spark-connector_2.13/1.2.3-spark-3.3.1/spark-connector_2.13-1.2.3-spark-3.3.1.pom (java.io.IOException: Impossible to load parent for file:/Users/winkjul/.ivy2/cache/com.exasol/spark-connector_2.13/ivy-1.2.3-spark-3.3.1.xml.original. Parent=com.exasol#spark-connector-generated-parent;1.2.3)

		module not found: com.exasol#spark-connector_2.13;1.2.3-spark-3.3.1

Any idea on how to fix this? I can install other dependencies without any issues for both Spark 3.2 and 3.3 it is just Exasol having issues.. 😢

**Comment by morazow** [comment]
Hello @julwin,

Thanks for the feedback! 

We are going to look into it. It seems like a bug on the released pom file, since the parent file is looked up.

Meanwhile please try to use the assembled jar file from releases if that is an option for you

**Comment by julwin** [comment]
Hey @morazow,

thanks for looking after this. Unfortunately the approach using the assembled jar file does not really work out yet. I'm trying to include the .jar, but keep getting "classNotFound" or "methodNotFound" errors..

**Comment by morazow** [comment]
Hello @julwin,

Thanks for the update!

Could you please share the all of the exception stacktrace?

And could you please tell which version of Spark are using? Please be aware that if the Spark version is for Scala 2.13, you should use that corresponding assemled jar.

**Comment by julwin** [comment]
Sure! One example would be when calling `df.write.mode("overwrite").format("exasol")...`:
```
org.apache.spark.sql.errors.QueryExecutionErrors$.failedToFindDataSourceError(QueryExecutionErrors.scala:587)
	at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSource(DataSource.scala:675)
	at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSourceV2(DataSource.scala:725)
	at org.apache.spark.sql.DataFrameWriter.lookupV2Provider(DataFrameWriter.scala:864)
	at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:256)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:247)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: exasol.DefaultSource
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$lookupDataSource$5(DataSource.scala:661)
	at scala.util.Try$.apply(Try.scala:213)
	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$lookupDataSource$4(DataSource.scala:661)
	at scala.util.Failure.orElse(Try.scala:224)
	at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSource(DataSource.scala:661)
	... 16 more
```


This is when adding the assembled jar via:

`conf.set(
            "spark.executor.extraClassPath",
            "home/airflow/.ivy2/jars/spark-connector_2.12-1.2.2-spark-3.2.2-assembly.jar",
        )`

Same for the `spark.jars` option, where the dependency solver complains that the directory is not allowed for addJar (tried different directories even)

**Comment by julwin** [comment]
Ok I have been able to resolve this.. I suspect some issue with fs permissions to be the underlying problem here. I am able to load the assembled jar now. 

The issue with the dependency solver persists, though.

**Comment by morazow** [comment]
Hello @julwin,

Thanks, that is good news!

Sure, we are going to check the maven release issue.

**Comment by morazow** [comment]
Hello @julwin,

We have released a new [version 1.2.4](https://github.com/exasol/spark-connector/releases/tag/1.2.4). It should fix the issue with referencing the artifact as dependency. 

Could you please check if it works? 

```
$ spark-shell --packages com.exasol:spark-connector_2.13:1.2.4-spark-3.3.2
:: loading settings :: url = jar:file:/opt/bitnami/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
Ivy Default Cache set to: /opt/bitnami/spark/.ivy2/cache
The jars for the packages stored in: /opt/bitnami/spark/.ivy2/jars
com.exasol#spark-connector_2.13 added as a dependency
:: resolving dependencies :: org.apache.spark#spark-submit-parent-66b7743d-235e-4774-ab24-e60b07651108;1.0
        confs: [default]
        found com.exasol#spark-connector_2.13;1.2.4-spark-3.3.2 in central
:: resolution report :: resolve 90ms :: artifacts dl 3ms
        :: modules in use:
        com.exasol#spark-connector_2.13;1.2.4-spark-3.3.2 from central in [default]
        ---------------------------------------------------------------------
        |                  |            modules            ||   artifacts   |
        |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
        ---------------------------------------------------------------------
        |      default     |   1   |   0   |   0   |   0   ||   1   |   0   |
        ---------------------------------------------------------------------
:: retrieving :: org.apache.spark#spark-submit-parent-66b7743d-235e-4774-ab24-e60b07651108
        confs: [default]
        0 artifacts copied, 1 already retrieved (0kB/5ms)
23/03/13 09:22:28 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
Spark context Web UI available at http://5d846bafa17b:4040
Spark context available as 'sc' (master = local[*], app id = local-1678699352128).
Spark session available as 'spark'.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 3.3.2
      /_/
         
Using Scala version 2.12.15 (OpenJDK 64-Bit Server VM, Java 1.8.0_362)
Type in expressions to have them evaluated.
Type :help for more information.

scala> 
``` 

I am going to close this ticket for now. Please feel free to open a new ticket if there are issues.
